rust/kernel/clk.rs

```diff
@@ -0,0 +1,54 @@
+// SPDX-License-Identifier: GPL-2.0
+
+//! Common clock framework.
+//!
+//! C header: [`include/linux/clk.h`](../../../../include/linux/clk.h)
+
+use crate::{bindings, device, error::Result, str::CStr, to_result};
+
+/// Represents `struct clk *`.
+///
+/// # Invariants
+///
+/// The pointer is valid.
+pub struct Clk(*mut bindings::clk);
+
+impl Clk {
+    /// Creates new clock structure from a raw pointer.
+    ///
+    /// # Safety
+    ///
+    /// The pointer must be valid.
+    pub unsafe fn new(clk: *mut bindings::clk) -> Self {
+        Self(clk)
+    }
+
+    /// Lookups and obtains a managed reference to a clock producer.
+    ///
+    /// * `dev` - device for clock "consumer".
+    /// * `id` - ID of clock consumer.
+    pub fn get(dev: &impl device::RawDevice, id: Option<&CStr>) -> Result<Self> {
```

Comment:
This is my interpretation that introduces a new variant of this function based on what has been said [previously](https://github.com/Rust-for-Linux/linux/pull/593#discussion_r767157499), specifically:

> Since this applies to all devices, I think it makes sense to move [a variant of] this function to be a method of `RawDevice`, so one could call it as `dev.clk_get(...)` instead of `clk::clk_get(dev, ...)`.

Besides, there is no argument for keeping this so it may be deleted.